lib/chef/chef_fs/file_system/repository/chef_repository_file_system_cookbook_dir.rb

```diff
@@ -97,9 +97,9 @@ def children
           end
 
           def can_have_child?(name, is_dir)
-            if is_dir
+            if is_dir && !%w{ root_files .. . }.include?(name)
```

Comment:
Should this be something like `!(root_files + %w{ .. . }).include?(name)` ? Looks like that array literal evaluates to `['root_files', '..', '.']` ?